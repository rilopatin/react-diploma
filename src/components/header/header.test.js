import React from 'react'
import Header from './header'
import {shallow} from 'enzyme'


describe('Testing <Header/>', () => {
    it('Header has been rendered correctly', () => {
        const header = shallow(<Header/>)
        expect(header).toMatchSnapshot();
    })
})