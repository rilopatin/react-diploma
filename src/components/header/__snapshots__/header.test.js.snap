// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Header/> Header has been rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="header"
      >
        <li
          className="header__item"
        >
          <Link
            replace={false}
            to="/"
          >
            <img
              alt="logo"
              src="Logo.svg"
            />
          </Link>
        </li>
        <li
          className="header__item"
        >
          <Link
            replace={false}
            to="/coffee/"
          >
            Our coffee
          </Link>
        </li>
        <li
          className="header__item"
        >
          <Link
            replace={false}
            to="/goods/"
          >
            For your pleasure
          </Link>
        </li>
      </ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="header__item"
          >
            <Link
              replace={false}
              to="/"
            >
              <img
                alt="logo"
                src="Logo.svg"
              />
            </Link>
          </li>,
          <li
            className="header__item"
          >
            <Link
              replace={false}
              to="/coffee/"
            >
              Our coffee
            </Link>
          </li>,
          <li
            className="header__item"
          >
            <Link
              replace={false}
              to="/goods/"
            >
              For your pleasure
            </Link>
          </li>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/"
            >
              <img
                alt="logo"
                src="Logo.svg"
              />
            </Link>,
            "className": "header__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt="logo"
                src="Logo.svg"
              />,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "src": "Logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/coffee/"
            >
              Our coffee
            </Link>,
            "className": "header__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Our coffee",
              "replace": false,
              "to": "/coffee/",
            },
            "ref": null,
            "rendered": "Our coffee",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/goods/"
            >
              For your pleasure
            </Link>,
            "className": "header__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "For your pleasure",
              "replace": false,
              "to": "/goods/",
            },
            "ref": null,
            "rendered": "For your pleasure",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="header"
        >
          <li
            className="header__item"
          >
            <Link
              replace={false}
              to="/"
            >
              <img
                alt="logo"
                src="Logo.svg"
              />
            </Link>
          </li>
          <li
            className="header__item"
          >
            <Link
              replace={false}
              to="/coffee/"
            >
              Our coffee
            </Link>
          </li>
          <li
            className="header__item"
          >
            <Link
              replace={false}
              to="/goods/"
            >
              For your pleasure
            </Link>
          </li>
        </ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="header__item"
            >
              <Link
                replace={false}
                to="/"
              >
                <img
                  alt="logo"
                  src="Logo.svg"
                />
              </Link>
            </li>,
            <li
              className="header__item"
            >
              <Link
                replace={false}
                to="/coffee/"
              >
                Our coffee
              </Link>
            </li>,
            <li
              className="header__item"
            >
              <Link
                replace={false}
                to="/goods/"
              >
                For your pleasure
              </Link>
            </li>,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/"
              >
                <img
                  alt="logo"
                  src="Logo.svg"
                />
              </Link>,
              "className": "header__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <img
                  alt="logo"
                  src="Logo.svg"
                />,
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "logo",
                  "src": "Logo.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/coffee/"
              >
                Our coffee
              </Link>,
              "className": "header__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Our coffee",
                "replace": false,
                "to": "/coffee/",
              },
              "ref": null,
              "rendered": "Our coffee",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/goods/"
              >
                For your pleasure
              </Link>,
              "className": "header__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "For your pleasure",
                "replace": false,
                "to": "/goods/",
              },
              "ref": null,
              "rendered": "For your pleasure",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
